systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -0,0 +1,134 @@
+        }
+        deployClusterOperatorWithEnvVars(extensionContext, testEnvVars);
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, kafkaReplicas).build());
+
+        LOGGER.info("Check for presence of ContainerPort 9090/tcp (tcp-ctrlplane) in first Kafka pod.");
+        final Pod kafkaPod = PodUtils.getPodsByPrefixInNameWithDynamicWait(NAMESPACE, clusterName + "-kafka-").get(0);
+        ContainerPort expectedControlPlaneContainerPort = new ContainerPort(9090, null, null, "tcp-ctrlplane", "TCP");
+        List<ContainerPort> kafkaPodPorts = kafkaPod.getSpec().getContainers().get(0).getPorts();
+        assertTrue(kafkaPodPorts.contains(expectedControlPlaneContainerPort));
+
+        LOGGER.info("Try to send some messages to Kafka over the next {} seconds.", messageCount);
+        KafkaTopic kafkaTopic = KafkaTopicTemplates.topic(clusterName, topicName)
+                .editSpec()
+                    .withReplicas(kafkaReplicas)
+                    .withPartitions(kafkaReplicas)
+                .endSpec()
+                .build();
+        resourceManager.createResource(extensionContext, kafkaTopic);
+
+        KafkaBasicExampleClients kafkaBasicClientJob = new KafkaBasicExampleClients.Builder()
+                .withProducerName(producerName)
+                .withConsumerName(consumerName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+                .withTopicName(topicName)
+                .withMessageCount(messageCount)
+                .withDelayMs(500)
+                .withNamespaceName(NAMESPACE)
+                .build();
+
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi().build());
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.consumerStrimzi().build());
+        JobUtils.waitForJobRunning(consumerName, NAMESPACE, Duration.ofSeconds(20).toMillis());
+
+        LOGGER.info("Delete first found Kafka broker pod.");
+        kubeClient(NAMESPACE).deletePod(NAMESPACE, kafkaPod);
+        StatefulSetUtils.waitForAllStatefulSetPodsReady(KafkaResources.kafkaStatefulSetName(clusterName), kafkaReplicas);
+
+        LOGGER.info("Force Rolling Update of Kafka via annotation.");
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(clusterName, k -> k.getMetadata().setAnnotations(Map.of(Annotations.ANNO_STRIMZI_IO_MANUAL_ROLLING_UPDATE, "true")), NAMESPACE);
+        // Wait some time for Strimzi Operator reconciliation to happen (so annotation is parsed and evaluated).
+        LOGGER.info("Wait {} ms for next reconciliation", Constants.RECONCILIATION_INTERVAL);
+        Thread.sleep(Constants.RECONCILIATION_INTERVAL);
+        StatefulSetUtils.waitForAllStatefulSetPodsReady(NAMESPACE, KafkaResources.kafkaStatefulSetName(clusterName), kafkaReplicas, ResourceOperation.getTimeoutForResourceReadiness(Constants.STATEFUL_SET));
```

Comment:
Why don't you wait for pods rolling update? You just opened a door for race condition by this code I think.

Reply:
`StatefulSetUtils.waitTillSsHasRolled`

Reply:
I did not know about that fucntion. Going to use it. thanks!